hyperparameter_train_network.py

```diff
@@ -41,69 +46,144 @@ def color_cat(s):
+    state_input = Input(shape=(7*num_blocks,),name='state_input')
+    words_input = Input(shape=(MAX_LEN,N_SYMBOLS), name='words_input')
+    words_lstm=LSTM(config.lstm_size)(words_input)
+    words_lstm = Dense(128, activation='relu')(words_lstm)
+
+    network_input = keras.layers.concatenate([words_lstm, state_input])
+
+    main_network = BuildNet(config.both_layers, network_input)
+    main_output = Dense(config.num_blocks, activation='softmax', name='main_output')(main_network)
+
+    return Model(
+        inputs=[state_input,words_input],
+        outputs=[main_output]
+    )
 
 ##LOAD ALL THE DATA
-df_in = pd.read_csv('C:/Users/bubulkr/rhit-xprize-neural-network/neural_in.csv')
-df_out= pd.read_csv('C:/Users/bubulkr/rhit-xprize-neural-network/neural_out.csv')
-train_points_in=df_in.iloc[:16000,15:17].values
-train_words_in=df_in.iloc[:16000,14:15].values[:,0]
-train_words_in=tokenize(train_words_in);
-train_state_in=df_in.iloc[:16000,:14]
-train_block_id_out=df_out.iloc[:16000,0].values
-train_points_out = df_out.iloc[:16000,1:3].values
-train_state_in = encode_state(train_state_in)
-train_state_in = train_state_in.values
-test_points_in=df_in.iloc[16000:,15:17].values
-test_words_in=df_in.iloc[16000:,14:15].values[:,0]
-test_words_in=tokenize(test_words_in);
-test_state_in=df_in.iloc[16000:,:14]
-test_block_id_out=df_out.iloc[16000:,0].values
-test_points_out = df_out.iloc[16000:,1:3].values
-test_state_in= encode_state(test_state_in)
-test_state_in = test_state_in.values
-
-
-##train
-final_network.fit({'state_input':train_state_in, 'words_input':train_words_in, 'position_input':train_points_in},{'block_id_output':train_block_id_out,'point_output':train_points_out},epochs=10,batch_size=50)
\ No newline at end of file
+df_in = pd.read_csv('./neural_in.csv', header=None)
+df_out= pd.read_csv('./neural_out.csv', header=None)
+
+num_samples = df_in.shape[0]
+num_samples_training = int(num_samples * 0.25)
+
+train_words_in = df_in.iloc[:num_samples_training,-1].values
+
+train_words_in = tokenize(train_words_in)
+train_state_in = df_in.iloc[:num_samples_training,:-1]
+train_state_in = encode_state(train_state_in.values)
+
+test_words_in = df_in.iloc[num_samples_training:,-1].values
+test_words_in = tokenize(test_words_in);
+test_state_in = df_in.iloc[num_samples_training:,:-1]
+test_state_in = encode_state(test_state_in.values)
+
+train_main_out = df_out.iloc[:num_samples_training,]
+test_main_out = df_out.iloc[num_samples_training:,]
+
+def main():
+    epochs = 10
+    num_iterations = 10
+    num_blocks = 10
+
+    best_result = None
+    best_network = None
+    for x in range(num_iterations):
+        config = NetworkConfig.random_network_config(num_blocks)
+        final_network = create_network(num_blocks, MAX_LEN, N_SYMBOLS)
+        final_network.compile(
+            loss='mean_squared_error',
+            optimizer='Adam',
+            metrics=['accuracy']
+        )
+
+        print('main output shape:', train_main_out.shape)
```

Comment:
Is this a leftover print statement, or is this used to show progress through the loop?

Reply:
Leftover. With `verbose` set to true there is already a lot of printing going on. I didn't think very hard about cleaning up things like this because the training doesn't really work yet. It runs but it is still being debugged, I just wanted others to be able to work with this instead of having four people each individually make these same changes.